### PR TITLE
Add primereact Tree bindings

### PR DESCRIPTION
**Pull request by hugo-vrijswijk**
Not all bindings are added, but we can add them as we go. Also see https://primereact.org/tree/

image

### CODE REVIEW

File: prime-react/src/main/scala/react/primereact/Tree.scala [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  opaque type Id = String
+  object Id:
+    inline def apply(value: String): Id                  = value
+    extension (opaqueValue: Id) inline def value: String = opaqueValue
+
+  case class Node[A](
+    id:       Tree.Id,
+    data:     A,
+    label:    js.UndefOr[String] = js.undefined,
+    icon:     Css = Css.Empty,
```
Comment (rpiaggio) [inline prime-react/src/main/scala/react/primereact/Tree.scala:70]: Can we make `icon: FontAwesomeIcon | Image | String` as we have in other components?

(Maybe that should be a type alias)

Comment (hugo-vrijswijk) [reply prime-react/src/main/scala/react/primereact/Tree.scala:70]: I don't think adding anything other than a `String` actually works in the binding. The typescript types are 'any', but everything is just added as a class added to an element, not as an actual element. So `FontAwesomeIcon` (or any `VdomNode`) won't work

Comment (rpiaggio) [reply prime-react/src/main/scala/react/primereact/Tree.scala:70]: I haven't looked at the `CTreeNode` generated type, but in the TS definition of `TreeNode` the icon is defined as `icon?: IconType<TreeNode> | undefined;` and `export type IconType<ParentProps> = React.ReactNode | ((options: IconOptions<ParentProps>) => React.ReactNode);` (https://github.com/primefaces/primereact/blob/2f35bcd8c5c8802d984b9f7edc5d94be85939a57/components/lib/utils/utils.d.ts#L151).

In other places where there's an `IconType<?>` we use `icon: FontAwesomeIcon | Image | String` and call its extension method `toPrime` (or `toPrimeWithClass`) to conform to the facade.

Comment (hugo-vrijswijk) [reply prime-react/src/main/scala/react/primereact/Tree.scala:70]: Oops, yeah you're totally right, I'll add a type alias and use it here as well

**Review by rpiaggio — APPROVED** [review]
Nice, LGTM.

Thank you for the type alias.

**Pull request merged**